ops/build_release.sh

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+PROJECT_NAME="Blue Button API"
+BUILD_MESSAGE=""
+GITHUB_REPO="CMSgov/bluebutton-web-server"
+
+usage() {
+    cat <<EOF >&2
+Start a new $PROJECT_NAME release.
+
+Usage: $(basename "$0") [-ch] [-t previous-tag new-tag]
+
+Options:
+  -c    wait for confirmation before committing and pushing to GitHub
+  -h    print this help text and exit
+  -t    manually specify tags
+  -p    automatically push new tags to origin.
+EOF
+}
+
+CONFIRM=
+MANUAL_TAGS=
+AUTO_PUSH=
+while getopts ":chtp" opt; do
+    case "$opt" in
+        c)
+            CONFIRM=1
+            ;;
+        t)
+            MANUAL_TAGS=1
+            ;;
+	p)
+	    AUTO_PUSH=1
+	    ;;
+        h)
+            usage
+            exit 0
+            ;;
+        \?)
+            echo "Invalid option: -$OPTARG" >&2
+            exit 1
+            ;;
+    esac
+done
+
+shift $((OPTIND-1))
+
+if [ $# -lt 2 ] && [ -n "$MANUAL_TAGS" ]; then
+  usage
+  exit 1
+fi
+
+# fetch tags before any tag lookups so we have the most up-to-date list
+# and generate the correct next release number
+git fetch --tags
+
+if [ -n "$MANUAL_TAGS" ]; then
+  PREVTAG="$1"
+  NEWTAG="$2"
+  PREVRELEASENUM=${PREVTAG//^r/}
+  NEWRELEASENUM=${NEWTAG//^r/}
+else
+  PREVTAG=$(git tag | sort -n | tail -1)
+  if [ ! -n "$PREVTAG" ]; then
+      PREVTAG="r0"
+      PREVRELEASENUM=0
+  else
+      PREVRELEASENUM=$(git tag | grep '^r[0-9]' | sed 's/^r//' | sort -n | tail -1)
+  fi
+  NEWRELEASENUM=$(($PREVRELEASENUM + 1))
+  PREVTAG="r$PREVRELEASENUM"
+  NEWTAG="r$NEWRELEASENUM"
+fi
+
+git checkout -b "release-$NEWRELEASENUM"
+
+git tag -a -m"$PROJECT_NAME release $NEWTAG" -s "$NEWTAG"
+
+if [ -n "$AUTO_PUSH" ]; then
+    git push --tags
+
+    git push origin "release-$NEWRELEASENUM"
+fi
+
+echo "Release $NEWTAG created."
+echo
+echo "Hotfixes should be made against release-$NEWRELEASENUM"
```

Comment:
just to make sure we're all on the same page: The idea is a release is tagged and a branch is made, any hotfixes go onto that branch which is then deployed. Figuring out how to get the hotfix back into develop cleanly is a longer conversation.

Sound right?

Reply:
Yeah, that makes sense to me. Hopefully we won't have to do this often.

Reply:
This is definitely not the method of doing this that I'm familiar with, but this is totally fine.

Reply:
@samgensburg-gov what's your usual hotfix process?

Reply:
Usually I've done this by having a release and a develop branch. Basically, master would be the thing you have deployed, while develop is a working copy. When you want to hotfix something, you make a branch off of master (normally you would branch from develop) and apply the fix, then merge it into both master, causing a deployment, and develop.

Reply:
got it, basically what's in [a successful git branching model](http://nvie.com/posts/a-successful-git-branching-model/). I've used that and a variation of [the cactus model](https://barro.github.io/2016/02/a-succesful-git-branching-model-considered-harmful/). I found the cactus model much easier to manage, though I don't agree with all of it. I mostly like the guidlines of the stable main branch, short lived feature branches, and release branches that don't get merged in. It made rapid development with a good number of people easy, and helped with ci/cd.

We should keep this discussion going and decide on a stable flow to try out for awhile.

Reply:
Like I said, the decision is up to you. Whatever makes things easiest for you to manage. I just like making it so that deployments automatically occur and only occur from master.